Dockerfile

```diff
@@ -12,28 +14,58 @@ COPY ${DISTBALL} ${TMP_ARCHIVE}
 RUN mkdir -p ${TMP_DIR} && \
     tar xfvz ${TMP_ARCHIVE} --strip 1 -C ${TMP_DIR} && \
     # already create volume dir to later have correct rights
-    mkdir ${TMP_DIR}/data
-
-RUN apt-get update && \
-    apt-get install tini && \
+    mkdir ${TMP_DIR}/data && \
+    apt-get -qq update && \
+    apt-get install -y --no-install-recommends tini=0.18.0-1 && \
     cp /usr/bin/tini ${TMP_DIR}/bin/tini
 
 COPY docker/utils/startup.sh ${TMP_DIR}/bin/startup.sh
-RUN chmod +x -R ${TMP_DIR}/bin/
-RUN chmod 0775 ${TMP_DIR} ${TMP_DIR}/data
+RUN chmod +x -R ${TMP_DIR}/bin/ && \
+    chmod 0775 ${TMP_DIR} ${TMP_DIR}/data
 
 # Building prod image
-FROM eclipse-temurin:17-jre-focal as prod
+FROM eclipse-temurin:17-jre-focal@sha256:${BASE_SHA} as prod
+
+# leave unset to use the default value at the top of the file
+ARG BASE_SHA
+
+LABEL org.opencontainers.image.base.digest="${BASE_SHA}"
+LABEL org.opencontainers.image.base.name="docker.io/library/eclipse-temurin:17-jre-focal"
 
 # Building dev image
 FROM eclipse-temurin:17-jdk-focal as dev
-RUN echo "running DEV pre-install commands"
-RUN apt-get update
-RUN curl -sSL https://github.com/jvm-profiling-tools/async-profiler/releases/download/v1.7.1/async-profiler-1.7.1-linux-x64.tar.gz | tar xzv
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN echo "running DEV pre-install commands" && \
+    curl -sSL https://github.com/jvm-profiling-tools/async-profiler/releases/download/v1.7.1/async-profiler-1.7.1-linux-x64.tar.gz | tar xzv
 
 # Building application image
+# hadolint ignore=DL3006
 FROM ${APP_ENV} as app
 
+ARG VERSION=""
+ARG DATE=""
+ARG REVISION=""
+
+# OCI labels: https://github.com/opencontainers/image-spec/blob/main/annotations.md
+LABEL org.opencontainers.image.created="${DATE}"
+LABEL org.opencontainers.image.authors="zeebe@camunda.com"
+LABEL org.opencontainers.image.url="https://zeebe.io"
+LABEL org.opencontainers.image.documentation="https://docs.camunda.io/docs/self-managed/zeebe-deployment/"
+LABEL org.opencontainers.image.source="https://github.com/camunda/zeebe"
+LABEL org.opencontainers.image.version="${VERSION}"
+LABEL org.opencontainers.image.revision="${REVISION}"
+LABEL org.opencontainers.image.vendor="Camunda Services GmbH"
+LABEL org.opencontainers.image.licenses="(Apache-2.0 AND Zeebe-Community-1.1)"
```

Comment:
:thought_balloon: Technically speaking, this is not according to spec, `Zeebe-Community-1.1` is not a valid license-id, see https://spdx.github.io/spdx-spec/SPDX-license-expressions/#d1-overview and https://spdx.org/licenses/

Reply:
I know, but it seems wrong to me to mark it as _only_ Apache 2.0 licensed (since that's only a small portion of the code). I tried finding out what to do when your license is not there and couldn't really find anything. I assume it will simply not be linked in integration tools.

Reply:
I think that we are supposed to use `LicenseRef` but documentation is very sparse :shrug:

Reply:
Yeah, could be. It's rather unclear, but it seems very likely. Nice find :+1: